encode_utils/connection.py

```diff
@@ -219,10 +219,11 @@ def _set_dcc_mode(self, dcc_mode=False):
             except KeyError:
                 print("ERROR: You must supply the `dcc_mode` argument or set the environment variable DCC_MODE.")
                 sys.exit(-1)
-        dcc_mode = dcc_mode.lower().rstrip("/")
+        scheme = 'https://'
+        dcc_mode = dcc_mode.lower().rstrip("/").replace(scheme, '')
         if dcc_mode not in eu.DCC_MODES:
             # Assume dcc_mode is a valid demo host.
-            url = 'https://' + dcc_mode + '/'
+            url = scheme + dcc_mode
```

Comment:
I think the trailing `/` was added because in the post() function it creates urls by joining dcc_mode and profile_id (which doesn't start with `/`) https://github.com/StanfordBioinformatics/encode_utils/blob/de68ce695bd478feb24d961bfff7555a7b80e3f4/encode_utils/connection.py#L964 so it might be necessary. patch() might have a similar issue.

Reply:
I believe the url_join strips the slashes, if they exist, and adds them back in, so it doesn't matter if this line has them with/without:
https://github.com/StanfordBioinformatics/encode_utils/blob/de68ce695bd478feb24d961bfff7555a7b80e3f4/encode_utils/utils.py#L156

If I remember correctly there was some other edge case that doesn't get handled gracefully by the changed code, but I'd need to double-check. I don't think this is merge-worthy as is regardless.